Quite a crude attempt to remove update checks, but it should work.

Update checks also used to trigger popup donation window at startup,
but not anymore with these patches.

--- a/DisplayCAL/display_cal.py
+++ b/DisplayCAL/display_cal.py
@@ -3119,16 +3119,6 @@
         self.Bind(wx.EVT_MENU, self.help_support_handler, menuitem)
         menuitem = help.FindItemById(help.FindItem("bug_report"))
         self.Bind(wx.EVT_MENU, self.bug_report_handler, menuitem)
-        self.menuitem_app_auto_update_check = help.FindItemById(
-            help.FindItem("update_check.onstartup")
-        )
-        self.Bind(
-            wx.EVT_MENU,
-            self.app_auto_update_check_handler,
-            self.menuitem_app_auto_update_check,
-        )
-        menuitem = help.FindItemById(help.FindItem("update_check"))
-        self.Bind(wx.EVT_MENU, self.app_update_check_handler, menuitem)
 
         if sys.platform == "darwin":
             wx.GetApp().SetMacAboutMenuItemId(self.menuitem_about.GetId())
@@ -3257,7 +3247,6 @@
         self.menuitem_show_log.Check(bool(getcfg("log.show")))
         self.menuitem_log_autoshow.Enable(not bool(getcfg("log.show")))
         self.menuitem_log_autoshow.Check(bool(getcfg("log.autoshow")))
-        self.menuitem_app_auto_update_check.Check(bool(getcfg("update_check")))
 
     def init_controls(self):
         """Initialize the main window controls and their event handlers."""
@@ -4047,7 +4036,6 @@
             "untethered.max_delta.chroma",
             "untethered.min_delta",
             "untethered.min_delta.lightness",
-            "update_check",
             "webserver.portnumber",
             "whitepoint.visual_editor.bg_v",
             "whitepoint.visual_editor.b",
@@ -19474,19 +19462,6 @@
         app.frame.Show()
         app.process_argv(1)
         wx.CallAfter(app.frame.Raise)
-        # Check for updates if configured
-        if getcfg("update_check"):
-            # Give time for the main window to gain focus before checking for
-            # update, otherwise the main window may steal the update
-            # confirmation dialog's focus which looks weird
-            wx.CallAfter(app.frame.app_update_check_handler, None, silent=True)
-        else:
-            # Check if we need to run instrument setup
-            wx.CallAfter(
-                app.frame.check_instrument_setup,
-                check_donation,
-                (app.frame, VERSION > VERSION_BASE),
-            )
         # If resources are missing, XRC shows an error dialog which immediately
         # gets hidden when we close ourselves because we are the parent.
         # Hide instead.
--- a/DisplayCAL/worker.py
+++ b/DisplayCAL/worker.py
@@ -1610,10 +1610,6 @@
         dlg_result = dlg.ShowModal()
         dlg.Destroy()
         if dlg_result == wx.ID_OK:
-            # Download Argyll CMS
-            from DisplayCAL.display_cal import app_update_check
-
-            app_update_check(parent, silent, argyll=True)
             return False
         elif dlg_result == wx.ID_CANCEL:
             if callafter:
--- a/DisplayCAL/xrc/mainmenu.xrc
+++ b/DisplayCAL/xrc/mainmenu.xrc
@@ -231,14 +231,6 @@
         <label>bug_report</label>
       </object>
       <object class="separator"/>
-      <object class="wxMenuItem" name="update_check.onstartup">
-        <label>update_check.onstartup</label>
-        <checkable>1</checkable>
-      </object>
-      <object class="wxMenuItem" name="update_check">
-        <label>update_check</label>
-      </object>
-      <object class="separator"/>
       <label>menu.help</label>
     </object>
   </object>
